### PR TITLE
LPS-177741 | LPS-177745 Integration Tests for Object Related Stories

### PR DESCRIPTION
**Pull request by jkappler**
Dear Team, this PR contains integration tests for the following stories

[LPS-161646 Page builder to support relationship tables](https://issues.liferay.com/browse/LPS-161646)
[LPS-174863 Enable users to map related item properties from the relationship fields to fragments](https://issues.liferay.com/browse/LPS-174863)

After this, we could close them pending QA review. Please let me know what you think and if changes should be made.

Thanks as always for all the help!

### CODE REVIEW

**Comment by jkappler** [comment]
I'll take a look at the failing test

**Comment by jkappler** [comment]
ci:test:relevant

**Comment by jkappler** [comment]
@MarinhoFeliphe, I don't like the last commit too much, but it seems to be needed because otherwise dataguard will fail when deleting the objectEntries 😞

File: modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/web/internal/info/item/provider/test/ObjectEntryInfoItemFieldValuesProviderTest.java [hunk 1]
```diff
@@ -135,18 +149,74 @@ public void testObjectEntryInfoItemFieldValuesProviderWithAttachment()
...
+		throws Exception {
+
+		return _objectRelationshipLocalService.addObjectRelationship(
+			TestPropsValues.getUserId(),
+			relatedObjectDefinition.getObjectDefinitionId(),
+			objectDefinition.getObjectDefinitionId(), 0,
+			ObjectRelationshipConstants.DELETION_TYPE_CASCADE,
+			LocalizedMapUtil.getLocalizedMap(RandomTestUtil.randomString()),
+			StringUtil.randomId(), type);
+	}
```
Comment (MarinhoFeliphe) [inline modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/web/internal/info/item/provider/test/ObjectEntryInfoItemFieldValuesProviderTest.java:214]: same here

File: modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/web/internal/info/item/provider/test/ObjectEntryInfoItemFieldValuesProviderTest.java [hunk 2]
```diff
@@ -135,18 +149,74 @@ public void testObjectEntryInfoItemFieldValuesProviderWithAttachment()
...
+
+	private void _assertAttachmentInfoFieldValues(
```
Comment (MarinhoFeliphe) [inline modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/web/internal/info/item/provider/test/ObjectEntryInfoItemFieldValuesProviderTest.java:216]: same here

File: modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/internal/info/collection/provider/test/OneToManyObjectRelationshipInfoCollectionProviderTest.java [hunk 3]
```diff
@@ -0,0 +1,246 @@
...
+	@Before
+	public void setUp() throws Exception {
+		PropsUtil.addProperties(
+			UnicodePropertiesBuilder.setProperty(
+				"feature.flag.LPS-176083", "true"
+			).build());
+
+		_group = GroupTestUtil.addGroup();
+
+		_objectDefinition = _addObjectDefinition(
```
Comment (MarinhoFeliphe) [inline modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/internal/info/collection/provider/test/OneToManyObjectRelationshipInfoCollectionProviderTest.java:84]: I believe we can rename to `childObjectDefinition`

File: modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/internal/info/collection/provider/test/OneToManyObjectRelationshipInfoCollectionProviderTest.java [hunk 4]
```diff
@@ -0,0 +1,246 @@
...
+			new TextObjectFieldBuilder(
+			).labelMap(
+				LocalizedMapUtil.getLocalizedMap(RandomTestUtil.randomString())
+			).name(
+				"name"
```
Comment (MarinhoFeliphe) [inline modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/internal/info/collection/provider/test/OneToManyObjectRelationshipInfoCollectionProviderTest.java:89]: here we can rename to `childTextObjectFieldName`

File: modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/internal/info/collection/provider/test/OneToManyObjectRelationshipInfoCollectionProviderTest.java [hunk 5]
```diff
@@ -0,0 +1,246 @@
...
+			_objectDefinitionLocalService.publishCustomObjectDefinition(
+				TestPropsValues.getUserId(),
+				_objectDefinition.getObjectDefinitionId());
+
+		_parentObjectDefinition = _addObjectDefinition(
+			new TextObjectFieldBuilder(
+			).labelMap(
+				LocalizedMapUtil.getLocalizedMap(RandomTestUtil.randomString())
+			).name(
+				"parentName"
```
Comment (MarinhoFeliphe) [inline modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/internal/info/collection/provider/test/OneToManyObjectRelationshipInfoCollectionProviderTest.java:104]: and here `parentTextObjectFieldName` to follow the pattern

File: modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/internal/info/collection/provider/test/OneToManyObjectRelationshipInfoCollectionProviderTest.java [hunk 6]
```diff
@@ -0,0 +1,246 @@
...
+		throws Exception {
+
+		return _objectRelationshipLocalService.addObjectRelationship(
+			TestPropsValues.getUserId(),
+			relatedObjectDefinition.getObjectDefinitionId(),
+			objectDefinition.getObjectDefinitionId(), 0,
+			ObjectRelationshipConstants.DELETION_TYPE_CASCADE,
+			LocalizedMapUtil.getLocalizedMap(RandomTestUtil.randomString()),
+			StringUtil.randomId(), type);
+	}
```
Comment (MarinhoFeliphe) [inline modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/internal/info/collection/provider/test/OneToManyObjectRelationshipInfoCollectionProviderTest.java:223]: same here

File: modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/internal/info/collection/provider/test/ManyToManyObjectRelationshipInfoCollectionProviderTest.java [hunk 7]
```diff
@@ -0,0 +1,262 @@
...
+	@Before
+	public void setUp() throws Exception {
+		PropsUtil.addProperties(
+			UnicodePropertiesBuilder.setProperty(
+				"feature.flag.LPS-176083", "true"
+			).build());
+
+		_group = GroupTestUtil.addGroup();
+
+		_relatedObjectDefinition = _addObjectDefinition(
```
Comment (MarinhoFeliphe) [inline modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/internal/info/collection/provider/test/ManyToManyObjectRelationshipInfoCollectionProviderTest.java:85]: I believe in this case we can rename to `objectDefinition1`

File: modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/internal/info/collection/provider/test/ManyToManyObjectRelationshipInfoCollectionProviderTest.java [hunk 8]
```diff
@@ -0,0 +1,262 @@
...
+			).objectFieldSettings(
+				Collections.emptyList()
+			).build());
+
+		_relatedObjectDefinition =
+			_objectDefinitionLocalService.publishCustomObjectDefinition(
+				TestPropsValues.getUserId(),
+				_relatedObjectDefinition.getObjectDefinitionId());
+
+		_objectDefinition = _addObjectDefinition(
```
Comment (MarinhoFeliphe) [inline modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/internal/info/collection/provider/test/ManyToManyObjectRelationshipInfoCollectionProviderTest.java:100]: so we can rename to `objectDefinition2`

File: modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/internal/info/collection/provider/test/ManyToManyObjectRelationshipInfoCollectionProviderTest.java [hunk 9]
```diff
@@ -0,0 +1,262 @@
...
+			).build());
+
+		_group = GroupTestUtil.addGroup();
+
+		_relatedObjectDefinition = _addObjectDefinition(
+			new TextObjectFieldBuilder(
+			).labelMap(
+				LocalizedMapUtil.getLocalizedMap(RandomTestUtil.randomString())
+			).name(
+				"relatedName"
```
Comment (MarinhoFeliphe) [inline modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/internal/info/collection/provider/test/ManyToManyObjectRelationshipInfoCollectionProviderTest.java:90]: here we can rename to `objectDefinition1TextObjectFieldName`

File: modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/internal/info/collection/provider/test/ManyToManyObjectRelationshipInfoCollectionProviderTest.java [hunk 10]
```diff
@@ -0,0 +1,262 @@
...
+			_objectDefinitionLocalService.publishCustomObjectDefinition(
+				TestPropsValues.getUserId(),
+				_relatedObjectDefinition.getObjectDefinitionId());
+
+		_objectDefinition = _addObjectDefinition(
+			new TextObjectFieldBuilder(
+			).labelMap(
+				LocalizedMapUtil.getLocalizedMap(RandomTestUtil.randomString())
+			).name(
+				"name"
```
Comment (MarinhoFeliphe) [inline modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/internal/info/collection/provider/test/ManyToManyObjectRelationshipInfoCollectionProviderTest.java:105]: and here objectDefinition2TextObjectFieldName

File: modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/internal/info/collection/provider/test/ManyToManyObjectRelationshipInfoCollectionProviderTest.java [hunk 11]
```diff
@@ -0,0 +1,262 @@
...
+		throws Exception {
+
+		return _objectRelationshipLocalService.addObjectRelationship(
+			TestPropsValues.getUserId(),
+			relatedObjectDefinition.getObjectDefinitionId(),
+			objectDefinition.getObjectDefinitionId(), 0,
+			ObjectRelationshipConstants.DELETION_TYPE_CASCADE,
+			LocalizedMapUtil.getLocalizedMap(RandomTestUtil.randomString()),
+			StringUtil.randomId(), type);
+	}
```
Comment (MarinhoFeliphe) [inline modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/internal/info/collection/provider/test/ManyToManyObjectRelationshipInfoCollectionProviderTest.java:239]: same here

File: modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/internal/info/collection/provider/test/ManyToManyObjectRelationshipInfoCollectionProviderTest.java [hunk 12]
```diff
@@ -0,0 +1,262 @@
...
+	@Test
+	public void testManyToManyObjectRelationshipRelatedInfoCollectionProvider()
+		throws Exception {
+
+		ObjectRelationship objectRelationship = _addObjectRelationship(
+			_relatedObjectDefinition, _objectDefinition,
+			ObjectRelationshipConstants.TYPE_MANY_TO_MANY);
+
+		try {
+			ObjectEntry relatedObjectEntry1 =
```
Comment (MarinhoFeliphe) [inline modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/internal/info/collection/provider/test/ManyToManyObjectRelationshipInfoCollectionProviderTest.java:133]: rename `relatedObjectEntry1`, `relatedObjectEntry2` and `objectEntry` to `objectDefinition1ObjectEntry1`, `objectDefinition1ObjectEntry2` and `objectDefinition2ObjectEntry`

File: modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/internal/info/collection/provider/test/ManyToManyObjectRelationshipInfoCollectionProviderTest.java [hunk 13]
```diff
@@ -0,0 +1,262 @@
...
+
+			Assert.assertEquals(2, collectionInfoPage.getTotalCount());
+
+			Assert.assertTrue(objectEntries.contains(relatedObjectEntry1));
+			Assert.assertTrue(objectEntries.contains(relatedObjectEntry2));
+		}
+		finally {
+			if (objectRelationship != null) {
+				_objectRelationshipLocalService.deleteObjectRelationship(
+					objectRelationship);
```
Comment (MarinhoFeliphe) [inline modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/internal/info/collection/provider/test/ManyToManyObjectRelationshipInfoCollectionProviderTest.java:209]: use @DeleteAfterTestRun to delete the `objectRelationship`

**Review by MarinhoFeliphe — CHANGES_REQUESTED** [review]
@jkappler 

does covering the other side of the relationship in ManyToManyObjectRelationshipInfoCollectionProviderTest make sense?

for example:

[here](https://github.com/liferay-objects/liferay-portal/pull/1884/files#diff-3aef1f623cfa412e3362177c03ffd66b74ddd20e39bea707d50b00cb87858306R188) it's being asserted the entries related to the objectEntry, but do you think you could do the same assert  for one of the other related entry, for example:
````java
collectionQuery.setRelatedItemObject(relatedObjectEntry1);

...

Assert.assertEquals(1, size());

Assert.assertEquals(1, collectionInfoPage.getTotalCount());

Assert.assertTrue(objectEntries.contains(objectEntry));
````

Thank you so much for your support!! :tada:

File: modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/web/internal/info/item/provider/test/ObjectEntryInfoItemFormProviderTest.java [hunk 14]
```diff
@@ -115,6 +127,42 @@ public void testObjectEntryInfoItemFormProviderWithAttachment()
...
+		_objectRelationshipLocalService.addObjectRelationship(
+			TestPropsValues.getUserId(),
+			relatedObjectDefinition.getObjectDefinitionId(),
+			objectDefinition.getObjectDefinitionId(), 0,
+			ObjectRelationshipConstants.DELETION_TYPE_CASCADE,
+			LocalizedMapUtil.getLocalizedMap(RandomTestUtil.randomString()),
+			StringUtil.randomId(), type);
+	}
+
+	private void _assertAttachmentInfoFields(InfoForm infoForm)
```
[the remark this answers is not in the document]
Comment (jkappler) [reply modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/web/internal/info/item/provider/test/ObjectEntryInfoItemFormProviderTest.java:163]: Hi @MarinhoFeliphe, we can do that, normally we are asked to separate the tests in two, that's why I did it. If it's no problem I will move it

File: modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/internal/info/collection/provider/test/ManyToManyObjectRelationshipInfoCollectionProviderTest.java (hunk 13 above)
Comment (jkappler) [reply modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/internal/info/collection/provider/test/ManyToManyObjectRelationshipInfoCollectionProviderTest.java:209]: Hi @MarinhoFeliphe, I tried this, but as I mentioned in my last commit, DataGuard throws an error when deleting the objects

**Comment by jkappler** [comment]
Hi @MarinhoFeliphe, pushed changes based on your comments, thanks a lot for reviewing this!

I hope that the ManyToMany test won't fail, since as I mentioned, the first time it failed because of an issue with Dataguard that was deleting the object entries after the definition was deleted in the tear down

**Comment by jkappler** [comment]
@MarinhoFeliphe, hope I didn't leave anything out 😅

**Comment by jkappler** [comment]
ci:test:relevant

**Comment by jkappler** [comment]
ci:forward

**Comment by jkappler** [comment]
sorry, tried to fw wrong PR

**Comment by jkappler** [comment]
ci:close

**Comment by jkappler** [comment]
ci:reopen

**Comment by jkappler** [comment]
@MarinhoFeliphe, ManyToMany failed again because of what I mentioned

**Comment by MarinhoFeliphe** [comment]
Resent here: https://github.com/liferay-objects/liferay-portal/pull/1902

@jkappler thanks for your changes!

**Comment by jkappler** [comment]
thanks to you @MarinhoFeliphe!

**Pull request closed without merging**